types/chain_info.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// ChainInfo is used to track metadata about a peer and its chain.
+type ChainInfo struct {
+	Peer    peer.ID
+	Head    TipSetKey
+	Height  uint64
+	Trusted bool
```

Comment:
I don't think the `hello` to a single node should indicate trustworthiness. This is a step we will have to revert when we do a more principled bootstrapping. The bootstrapper/peertracker will compute a trusted head from *many* peers (though it's ok in the interim to "trust" all of them).

In @frrist's downstream changes, I think it's fine that this results in two arguments to the syncer's HandleNewTipSet.

Reply:
Got it.  Since I did this to make @frrist's rebase easier I'll leave it up to him in #3167

Reply:
I'm good with taking it out of `ChainInfo`, conflicts should be manageable.